### ISSUE TITLE
Different result despite same input

### ISSUE THREAD

**Issue by iamhuy** [TeamHG-Memex/sklearn-crfsuite#9]
I tried to create some CRF instances to train with the same training set and same max_iteration param.
```
crf = sklearn_crfsuite.CRF(
            algorithm='ap', 
            max_iterations=5, 
        )
crf.fit(X_train, Y_train)

t = sklearn_crfsuite.CRF(
            algorithm='ap', 
            max_iterations=5, 
        )
t.fit(X_train, Y_train)
```
However, their result is different ( I tested them on the same develop set with fmeasure).
Hope to see your response soon.
Thank you

**Comment by kmike** [comment]
I think this is expected - crfsuite [shuffles](https://github.com/chokkan/crfsuite/blob/8c0028c5070546a5fc08d2c8175ab244c618f35f/lib/crf/src/train_averaged_perceptron.c#L159) dataset for Averaged Perceptron training, and uses a global random seed (see [here](https://github.com/chokkan/crfsuite/blob/8c0028c5070546a5fc08d2c8175ab244c618f35f/lib/crf/src/dataset.c#L92)); it means shuffle returns a different result each time.

**Comment by iamhuy** [comment]
Thank you ! 
Does it mean for a specific set of hyperparameters, it is necessary to train more than one time to find the best one ( because it depends on the time too) ?

**Comment by kmike** [comment]
Well, it depends on a goal. If you want to compare hyperparameters then yeah, it could make sense to train on several seeds, and take e.g. an average, or a best model, or just compute variance. But are results really that different in different runs?

**Comment by iamhuy** [comment]
No. They're not different on different runs:
I mean if I run above code with in 2 different execitions that crf1, t1 ,crf2, t2:
then crf1 = crf2 , t1 = t2 and crf1 != t1

**Comment by severinsimmler** [comment]
How do I set the random seed?

**Comment by huang-xx** [comment]
@iamhuy  @severinsimmler   Hi, I encountered the same problem, but after I set` random_state` for the `train_test_split` function of `sklearn.model_selection`, the results became consistent.

**Comment by UAmsterdam** [comment]
I am also getting different results while running it on different environment:
 Like command-line verison of CRFsuite and Python verison of CRFsuite. 
 
 Does anyone here has some idea whats going on?
